### PR TITLE
empty hash to hosts variable

### PR DESCRIPTION
**Pull request by dal00**
We need an empty hash here to not fail in case we have hostsdefaults defined in hiera but no hosts.

### CODE REVIEW

**Comment by buzzdeee** [comment]
I wonder how much sense does it make to include the class somewhere, and have hostsdefaults defined in hiera, but no hosts?
I had it with no default by intent, to bail out early, i.e.  in case of typos in Hiera (happens for me from time to time ;)

If you convince me it's useful for something, let me know and I'll merge it ;)

Sebastian

**Comment by dal00** [comment]
The use case is basically that I have the defaults set for all sites we have in hiera via a "serverrole".
Then I set the hostname and mac only per site location. But if a site does not have hosts defined the merge will fail and bail out. I need to be able to have locations defined without it breaking because this is not in use at all locations.

hiera yaml:
  - "location/%{::serverlocation}" 
  - "serverrole/%{::serverrole}"

Did this make sense?

**Comment by buzzdeee** [comment]
Your example makes sense, but one last question, do you have a global.yaml or the like that is called last, and defines last resort defaults for all hosts?

You could define a 

autoinstall::hosts: {}

empty hash in that last resort global.yaml (or however you have called it ;) to get around it. So your locations which have/need hosts configured for autoinstall, will be fine, and the others will get the empty hash from that last resort default.

So basically you move the default value from the init.pp parameter to the last resort hiera file.

**Comment by dal00** [comment]
I do but we can't expect that everybody has that setup and it's very ugly to have an empty hash in hiera.
And if somebody would remove the empty hash from our production hiera because it's "not doing anything" it could break a lot of stuff.

**Comment by buzzdeee** [comment]
OK, I'm convinced now ;)
